Vimperator broken in latest Nightly 41.0a1 (2015-05-22)

It does not respond to any key shortcuts, and the theme, key bindings, and other customizations are not loaded, either. It is likely this is just an issue with today's build of Nightly, but in case the issue remains from this point forward, an issue should be documented.

FYI, this seems to be working for me too on windows 8.1 64bit / FF 41.0a2.

Couldn't build it under windows though, both master and fixup-nightly errors out with:

  $ make xpi
  Building XPI...
  Including en-US documentation
  Including ja documentation
  Replacing ###VERSION### and ###DATE### tags
  File not found - *.css
  make: *** [xpi] Error 2

This is under cygwin -- and the combination that got farthest along (tried mingw64 and the stuff that comes with git/bash). No problems building on Debian 7/old stable (the server that was easiest to try - didn't try on 8/stable -- but presumably builds fine there too). Probably should file a separate bug for instructions on how to build under windows -- assuming it's supposed to be supported at all? (A pointer to an installer/setup w/make++ that's expected to work would go a long way).

--- a/common/content/base.js
+++ b/common/content/base.js
@@ -290,7 +290,14 @@ function extend(subclass, superclass, overrides) {
  * @returns {function} The constructor for the resulting class.
  */
 function Class() {
-    function constructor() {
+    var args = Array.slice(arguments);
+    if (isstring(args[0]))
+        var name = args.shift();
+    var superclass = Class;
+    if (callable(args[0]))
+        superclass = args.shift();
+
+    let Constructor = eval(`(function ${(name || superclass.name).replace(/\W/g, '_')}() {
         let self = {
             __proto__: Constructor.prototype,
             constructor: Constructor,
@@ -305,17 +312,7 @@ function Class() {
         };
         var res = self.init.apply(self, arguments);
         return res !== undefined ? res : self;
-    }
-
-    var args = Array.slice(arguments);
-    if (isstring(args[0]))
-        var name = args.shift();
-    var superclass = Class;
-    if (callable(args[0]))
-        superclass = args.shift();
-
-    var Constructor = eval("(function " + (name || superclass.name).replace(/\W/g, "_") +
-            String.substr(constructor, 20) + ")");
+    })`);
     Constructor.__proto__ = superclass;
 
     if (!("init" in superclass.prototype) && !("init" in args[0])) {

--- a/common/content/bookmarks.js
+++ b/common/content/bookmarks.js
@@ -505,7 +505,7 @@ const Bookmarks = Module("bookmarks", {
                 let jumps = [[idx == sh.index ? ">" : "",
                               Math.abs(idx - sh.index),
                               val.title,
-                              let(url=val.URI.spec) xml`<a highlight="URL" href=${url}>${url}</a>`]
+                              xml`<a highlight="URL" href=${val.URI.spec}>${val.URI.spec}</a>`]
                               for ([idx, val] in Iterator(sh))];
                 let list = template.tabular([{ header: "Jump", style: "color: red", colspan: 2 },
                     { header: "", style: "text-align: right", highlight: "Number" },

--- a/common/content/buffer.js
+++ b/common/content/buffer.js
@@ -1354,9 +1354,13 @@ const Buffer = Module("buffer", {
             return " ";
         }
         function getURLFromTab (tab) {
-            if ("linkedBrowser" in tab)
-                return tab.linkedBrowser.contentDocument.location.href;
-            else {
+            if ("linkedBrowser" in tab) {
+                if (tab.linkedBrowser.contentDocument) {
+                    return tab.linkedBrowser.contentDocument.location.href;
+                } else if (tab.linkedBrowser.lastURI.asciiSpec) {
+                    return tab.linkedBrowser.lastURI.asciiSpec
+                }
+            } else {
                 let i = config.tabbrowser.mTabContainer.getIndexOfItem(tab);
                 let info = config.tabbrowser.tabInfo[i];
                 return info.browser ?

--- a/common/content/commandline.js
+++ b/common/content/commandline.js
@@ -11,6 +11,8 @@
  * It consists of a prompt and command field be sure to only create objects of
  * this class when the chrome is ready.
  */
+let timeID = null;
+
 const CommandLine = Module("commandline", {
     requires: ["config", "liberator", "modes", "services", "storage", "template", "util"],
 
@@ -286,7 +288,7 @@ const CommandLine = Module("commandline", {
      * @param {boolean} forceSingle If provided, don't let over-long
      *     messages move to the MOW.
      */
-    _echoLine: let (timeID = null) function (str, highlightGroup, forceSingle) {
+    _echoLine: function (str, highlightGroup, forceSingle) {
         if (timeID) {
             window.clearTimeout(timeID);
             timeID = null;
@@ -973,12 +975,11 @@ const CommandLine = Module("commandline", {
                     showHelp = true;
                 break;
             case "Y":
-                let (sel = win.getSelection().toString()) {
-                    if (sel)
-                        util.copyToClipboard(sel, false);
-                    else
-                        showHelp = true;
-                }
+                let sel = win.getSelection().toString();
+                if (sel)
+                    util.copyToClipboard(sel, false);
+                else
+                    showHelp = true;
                 break;
             // unmapped key -> show Help
             default:
@@ -1192,10 +1193,9 @@ const CommandLine = Module("commandline", {
             // Change the completion text.
             // The second line is a hack to deal with some substring
             // preview corner cases.
-            let (str = this.prefix + completion + this.suffix) {
-                commandline._commandWidget.value = str;
-                this.editor.selection.focusNode.textContent = str;
-            }
+            let str = this.prefix + completion + this.suffix;
+            commandline._commandWidget.value = str;
+            this.editor.selection.focusNode.textContent = str;
 
             // Reset the caret to one position after the completion.
             this.caret = this.prefix.length + completion.length;

--- a/common/content/events.js
+++ b/common/content/events.js
@@ -1164,6 +1164,9 @@ const Events = Module("events", {
 }, {
     isInputElemFocused: function () {
         let elem = liberator.focus;
+        if (!elem) {
+            return false;
+        }
         return ((elem instanceof HTMLInputElement && !/image/.test(elem.type)) ||
                  elem instanceof HTMLTextAreaElement ||
                  elem instanceof HTMLObjectElement ||

--- a/common/content/io.js
+++ b/common/content/io.js
@@ -272,7 +272,7 @@ const File = Class("File", {
         // TODO: Vim expands paths twice, once before checking for ~, once
         // after, but doesn't document it. Is this just a bug? --Kris
         path = expand(path);
-        return path.replace("/", IO.PATH_SEP, "g");
+        return path.replace(/\//g, IO.PATH_SEP);
     },
 
     getPathsFromPathList: function (list) {

--- a/common/content/liberator.js
+++ b/common/content/liberator.js
@@ -714,7 +714,8 @@ const Liberator = Module("liberator", {
         }
 
         liberator.log('Searching for "plugin/**/*.{js,vimp}" in "'
-                            + [dir.path.replace(/.plugin$/, "") for (dir of dirs)].join(",") + '"');
+                            + dirs.map(function(dir) { return dir.path.replace(/.plugins$/, ''); }).join(',')
+                            + '"');
 
         dirs.forEach(function (dir) {
             liberator.log("Searching for \"" + (dir.path + "/**/*.{js,vimp}") + "\"", 3);
@@ -1307,7 +1308,9 @@ const Liberator = Module("liberator", {
                 validator: function (value) {
                     let toolbars = config.toolbars || {};
                     // "ne" is a simple hack, since in the next line val.replace() makes "ne" out from "none"
-                    let values = ["all", "ne"].concat([toolbar for each([toolbar, ] in Iterator(toolbars))]);
+                    let values = ["all", "ne"].concat(Object.keys(toolbars).map(function(tl) {
+                        return tl[0];
+                    }));
                     return value.every(function(val) values.indexOf(val.replace(/^(no|inv)/, "")) >= 0);
                 }
             });

--- a/common/content/liberator.js
+++ b/common/content/liberator.js
@@ -9,8 +9,6 @@
 Cu.import("resource://gre/modules/XPCOMUtils.jsm", modules);
 Cu.import("resource://gre/modules/AddonManager.jsm")
 
-const util = Cu.util;
-
 const plugins = { __proto__: modules };
 const userContext = { __proto__: modules };
 

--- a/common/content/liberator.js
+++ b/common/content/liberator.js
@@ -9,6 +9,8 @@
 Cu.import("resource://gre/modules/XPCOMUtils.jsm", modules);
 Cu.import("resource://gre/modules/AddonManager.jsm")
 
+const util = Cu.util;
+
 const plugins = { __proto__: modules };
 const userContext = { __proto__: modules };
 

--- a/common/content/liberator.js
+++ b/common/content/liberator.js
@@ -1305,11 +1305,10 @@ const Liberator = Module("liberator", {
                 },
                 validator: function (value) {
                     let toolbars = config.toolbars || {};
-                    // "ne" is a simple hack, since in the next line val.replace() makes "ne" out from "none"
-                    let values = ["all", "ne"].concat(Object.keys(toolbars).map(function(tl) {
-                        return tl[0];
-                    }));
-                    return value.every(function(val) values.indexOf(val.replace(/^(no|inv)/, "")) >= 0);
+                    let opts = Object.keys(toolbars);
+                    return value.every(function(val) {
+                        return opts.indexOf(val.replace(/^(no|inv)/, "")) >= 0;
+                    });
                 }
             });
 

--- a/common/content/liberator.xul
+++ b/common/content/liberator.xul
@@ -102,7 +102,7 @@
                 <hbox id="liberator-commandline" class="liberator-container" liberator:highlight="CmdLine">
                     <label class="plain" id="liberator-commandline-prompt-text" liberator:highlight="PromptText" flex="0" crop="end" value="" collapsed="false"/>
                     <label class="plain" id="liberator-commandline-prompt" liberator:highlight="Prompt" flex="0" crop="none" value="" collapsed="false"/>
-                    <textbox class="plain" id="liberator-commandline-command" flex="1" type="timed" timeout="100"
+                    <textbox class="plain" id="liberator-commandline-command" flex="1" type="search" timeout="100"
                         oninput="window.liberator &and; liberator.modules.commandline.onEvent(event);"
                         onkeyup="window.liberator &and; liberator.modules.commandline.onEvent(event);"
                         onfocus="window.liberator &and; liberator.modules.commandline.onEvent(event);"

--- a/common/content/mappings.js
+++ b/common/content/mappings.js
@@ -347,10 +347,10 @@ const Mappings = Module("mappings", {
      * @param {RegExp|string} URL matching pattern or URL.
      * @returns {boolean}
      */
-    hasMap:
-        function (mode, cmd, patternOrUrl)
-            let (self = this)
-                this._user[mode].some(function (map) map.hasName(cmd) && self._matchingUrlsTest(map, patternOrUrl)),
+    hasMap: function (mode, cmd, patternOrUrl) {
+        let self = this;
+        this._user[mode].some(function (map) map.hasName(cmd) && self._matchingUrlsTest(map, patternOrUrl));
+    },
 
     /**
      * Remove the user-defined mapping named <b>cmd</b> for <b>mode</b>.

--- a/common/content/modes.js
+++ b/common/content/modes.js
@@ -132,7 +132,7 @@ const Modes = Module("modes", {
     get mainMode() this._modeMap[this._main],
 
     addMode: function (name, extended, options) {
-        let disp = name.replace("_", " ", "g");
+        let disp = name.replace(/_/g, " ");
         this[name] = 1 << this._lastMode++;
         if (typeof extended == "object") {
             options = extended;

--- a/common/content/options.js
+++ b/common/content/options.js
@@ -197,7 +197,7 @@ const Option = Class("Option", {
      * Returns whether the specified <b>values</b> are valid for this option.
      * @see Option#validator
      */
-    isValidValue: function (values) this.validator(values),
+    isValidValue: function (vs) this.validator(vs),
 
     /**
      * Resets the option to its default value.

--- a/common/content/options.js
+++ b/common/content/options.js
@@ -716,7 +716,7 @@ const Options = Module("options", {
      *     any of the options's names.
      */
     remove: function (name) {
-        for each (let option in this._optionHash) {
+        for (let option of this._optionHash) {
             if (option.hasName(name))
                 delete this._optionHash[option.name];
         }

--- a/common/content/style.js
+++ b/common/content/style.js
@@ -550,28 +550,27 @@ Module("styles", {
     requires: ["config", "liberator", "storage", "util"],
 
     init: function () {
-        let (array = util.Array) {
-            update(Styles.prototype, {
-                get sites() array([v.sites for ([k, v] in this.userSheets)]).flatten().uniq().__proto__,
-                completeSite: function (context, content) {
-                    context.anchored = false;
-                    try {
-                        context.fork("current", 0, this, function (context) {
-                            context.title = ["Current Site"];
-                            context.completions = [
-                                [content.location.host, "Current Host"],
-                                [content.location.href, "Current URL"]
-                            ];
-                        });
-                    }
-                    catch (e) {}
-                    context.fork("others", 0, this, function (context) {
-                        context.title = ["Site"];
-                        context.completions = [[s, ""] for (s of styles.sites)];
+        let array = util.Array;
+        update(Styles.prototype, {
+            get sites() array([v.sites for ([k, v] in this.userSheets)]).flatten().uniq().__proto__,
+            completeSite: function (context, content) {
+                context.anchored = false;
+                try {
+                    context.fork("current", 0, this, function (context) {
+                        context.title = ["Current Site"];
+                        context.completions = [
+                            [content.location.host, "Current Host"],
+                            [content.location.href, "Current URL"]
+                        ];
                     });
                 }
-            });
-        }
+                catch (e) {}
+                context.fork("others", 0, this, function (context) {
+                    context.title = ["Site"];
+                    context.completions = [[s, ""] for (s of styles.sites)];
+                });
+            }
+        });
         return storage.newObject("styles", Styles, { store: false });
     }
 }, {

--- a/common/content/style.js
+++ b/common/content/style.js
@@ -261,7 +261,7 @@ function Highlights(name, store) {
         }
 
         let css = newStyle.replace(/(?:!\s*important\s*)?(?:;?\s*$|;)/g, "!important;")
-                          .replace(";!important;", ";", "g"); // Seeming Spidermonkey bug
+                          .replace(/;!important;/g, ";"); // Seeming Spidermonkey bug
         if (!/^\s*(?:!\s*important\s*)?;*\s*$/.test(css)) {
             css = style.selector + " { " + css + " }";
 

--- a/common/content/template.js
+++ b/common/content/template.js
@@ -19,7 +19,7 @@ const Template = Module("template", {
         let ret = tag``;
         let n = 0;
         var op = tag["+="] || tag["+"] ||function (lhs, rhs) tag`${lhs}${rhs}`;
-        for each (let i in Iterator(iter)) {
+        for (let i of Iterator(iter)) {
             let val = func(i);
             if (val == undefined || (tag.isEmpty && tag.isEmpty(val)))
                 continue;

--- a/common/content/util.js
+++ b/common/content/util.js
@@ -18,6 +18,8 @@ const {XHTML, XUL, NS} = (function () {
     };
 })();
 
+let Encoder = Components.Constructor("@mozilla.org/layout/documentEncoder;1?type=text/plain", "nsIDocumentEncoder", "init");
+
 const Util = Module("util", {
     init: function () {
         this.Array = Util.Array;
@@ -779,8 +781,7 @@ const Util = Module("util", {
      * @param {Number} flags    nsIDocumentEncoder.OutputXXXX
      * @returns {String}
      */
-    domToStr: let (Encoder = Components.Constructor("@mozilla.org/layout/documentEncoder;1?type=text/plain", "nsIDocumentEncoder", "init"))
-    function domToStr(node, type, flags) {
+    domToStr: function domToStr(node, type, flags) {
         var doc, method;
 
         if (node instanceof Document) {

--- a/common/content/util.js
+++ b/common/content/util.js
@@ -138,7 +138,7 @@ const Util = Module("util", {
      * @returns {string}
      */
     escapeString: function escapeString(str, delimiter) {
-        if (delimiter == undefined)
+        if (delimiter === undefined)
             delimiter = '"';
         return delimiter + str.replace(/([\\'"])/g, "\\$1").replace("\n", "\\n", "g").replace("\t", "\\t", "g") + delimiter;
     },
@@ -258,9 +258,9 @@ const Util = Module("util", {
         zip.open(FILE, io.MODE_CREATE | io.MODE_WRONLY | io.MODE_TRUNCATE);
         function addURIEntry(file, uri)
             zip.addEntryChannel(PATH + file, TIME, 9,
-                services.get("io").newChannel(uri, null, null), false);
+                services.get("io").newChannel(uri, null, null), false)
         function addDataEntry(file, data) // Inideal to an extreme.
-            addURIEntry(file, "data:text/plain;charset=UTF-8," + encodeURI(data));
+            addURIEntry(file, "data:text/plain;charset=UTF-8," + encodeURI(data))
 
         let empty = util.Array.toObject(
             "area base basefont br col frame hr img input isindex link meta param"
@@ -286,25 +286,25 @@ const Util = Module("util", {
                             data.push(" xmlns=" + JSON.stringify(XHTML.uri));
 
                         for (let { name: name, value: value } in util.Array.itervalues(node.attributes)) {
-                            if (name == "liberator:highlight") {
+                            if (name === "liberator:highlight") {
                                 name = "class";
                                 value = "hl-" + value;
                             }
-                            if (name == "href") {
-                                if (value.indexOf("liberator://help-tag/") == 0)
+                            if (name === "href") {
+                                if (value.indexOf("liberator://help-tag/") === 0)
                                     value = services.get("io").newChannel(value, null, null).originalURI.path.substr(1);
                                 if (!/[#\/]/.test(value))
                                     value += ".xhtml";
                             }
-                            if (name == "src" && value.indexOf(":") > 0) {
-                                chrome[value] = value.replace(/.*\//, "");;
+                            if (name === "src" && value.indexOf(":") > 0) {
+                                chrome[value] = value.replace(/.*\//, "");
                                 value = value.replace(/.*\//, "");
                             }
                             data.push(" ");
                             data.push(name);
                             data.push('="');
                             data.push(xml`${value}`.toString());
-                            data.push('"')
+                            data.push('"');
                         }
                         if (node.localName in empty)
                             data.push(" />");
@@ -313,7 +313,9 @@ const Util = Module("util", {
                             if (node instanceof HTMLHeadElement)
                                 data.push(`<link rel="stylesheet" type="text/css" href="help.css"/>`/*toXMLString()*/);
                             Array.map(node.childNodes, arguments.callee);
-                            data.push("</"); data.push(node.localName); data.push(">");
+                            data.push("</");
+                            data.push(node.localName);
+                            data.push(">");
                         }
                         break;
                     case Node.TEXT_NODE:
@@ -357,7 +359,7 @@ const Util = Module("util", {
             xmlhttp.mozBackgroundRequest = true;
             if (callback) {
                 xmlhttp.onreadystatechange = function () {
-                    if (xmlhttp.readyState == 4)
+                    if (xmlhttp.readyState === 4)
                         callback(xmlhttp);
                 };
             }
@@ -514,7 +516,7 @@ const Util = Module("util", {
         ]);
         if (object instanceof Element) {
             let elem = object;
-            if (elem.nodeType == elem.TEXT_NODE)
+            if (elem.nodeType === elem.TEXT_NODE)
                 return elem.data;
             function namespaced(node) {
                 var ns = NAMESPACES[node.namespaceURI];
@@ -558,7 +560,7 @@ const Util = Module("util", {
                 }
                 catch (e) {}
                 if (!hasValue) {
-                    if (i instanceof Array && i.length == 2)
+                    if (i instanceof Array && i.length === 2)
                         [i, value] = i;
                     else
                         var noVal = true;
@@ -839,7 +841,7 @@ const Util = Module("util", {
         }
     }, {
         isinstance: function isinstance(obj) {
-            return Object.prototype.toString.call(obj) == "[object Array]";
+            return Object.prototype.toString.call(obj) === "[object Array]";
         },
 
         /**

--- a/common/skin/liberator.css
+++ b/common/skin/liberator.css
@@ -10,233 +10,224 @@
 }
 
 /* Applied only to completion buffer and MOW */
-@-moz-document
-    url-prefix(chrome://liberator/) {
+@-moz-document url-prefix(chrome://liberator/) {
 
-*:-moz-loading, *:-moz-broken { display: none !important; }
+    *:-moz-loading, *:-moz-broken { display: none !important; }
 
-[liberator|highlight~=Completions] {
-    width: 100%;
-    display: table;
-}
-[liberator|highlight~=CompItem],
-[liberator|highlight~=CompTitle] {
-    display: table-row;
-}
-[liberator|highlight~=Completions] > ul {
-    display: table-row;
-}
-[liberator|highlight~=CompItem] > *,
-[liberator|highlight~=CompTitle] > * {
-    /* disabled for now, since it throws SecurityErrors to the console */
-    /* -moz-binding: url(chrome://liberator/content/bindings.xml#compitem-td); */
-    display: table-cell;
-    vertical-align: middle;
-}
+    [liberator|highlight~=Completions] {
+        width: 100%;
+        display: table;
+    }
+    [liberator|highlight~=CompItem],
+    [liberator|highlight~=CompTitle] {
+        display: table-row;
+    }
+    [liberator|highlight~=Completions] > ul {
+        display: table-row;
+    }
+    [liberator|highlight~=CompItem] > *,
+    [liberator|highlight~=CompTitle] > * {
+        /* disabled for now, since it throws SecurityErrors to the console */
+        /* -moz-binding: url(chrome://liberator/content/bindings.xml#compitem-td); */
+        display: table-cell;
+        vertical-align: middle;
+    }
 
-[liberator|highlight~=CompMsg] {
-    height: 1.5em;
-    line-height: 1.5em !important;
-}
+    [liberator|highlight~=CompMsg] {
+        height: 1.5em;
+        line-height: 1.5em !important;
+    }
 
-.td-span {
-    display: inline-block;
-    overflow: visible;
-    width: 0px;
-    height: 1.5em;
-    line-height: 1.5em !important;
-}
-.td-strut {
-    display: inline-block;
-    vertical-align: middle;
-    height: 16px;
-    width: 0px;
-}
+    .td-span {
+        display: inline-block;
+        overflow: visible;
+        width: 0px;
+        height: 1.5em;
+        line-height: 1.5em !important;
+    }
+    .td-strut {
+        display: inline-block;
+        vertical-align: middle;
+        height: 16px;
+        width: 0px;
+    }
 
-.extra-info { color: gray; }
+    .extra-info { color: gray; }
 
-.times-executed, .time-average { color: green; }
-.time-total { color: red; }
+    .times-executed, .time-average { color: green; }
+    .time-total { color: red; }
 
 }
 
 /* Applied to completion buffer, MOW, browser window */
-@-moz-document
-    url-prefix(chrome://) {
+@-moz-document url-prefix(chrome://) {
 
-.liberator-container > * {
-    font-family: inherit;
-}
+    .liberator-container > * {
+        font-family: inherit;
+    }
 
-.liberator-floatbox {
-    position: fixed;
-    width: 100%;
-    height: 100%;
-    z-index: 1000;
-}
+    .liberator-floatbox {
+        position: fixed;
+        width: 100%;
+        height: 100%;
+    }
 
-.liberator-container.animation[collapsed='true'] {
-    opacity: 0;
-    pointer-events: none;
-    max-height: 10px !important;
-    min-height: 10px;
-}
+    .liberator-container.animation[collapsed='true'] {
+        opacity: 0;
+        pointer-events: none;
+        max-height: 10px !important;
+        min-height: 10px;
+    }
 
-.liberator-floatbox #liberator-completions,
-.liberator-floatbox #liberator-multiline-output {
-    background-color: transparent;
-}
+    .liberator-floatbox > iframe {
+        height: 100%;
+        width: 100%;
+    }
 
-.liberator-floatbox > iframe {
-    height: 100%;
-    width: 100%;
-}
+    .liberator-container.animation {
+        transition: all 0.2s;
+    }
 
-.liberator-container.animation {
-    transition: all 0.2s;
-}
+    #liberator-completions {
+        -moz-user-focus: ignore;
+        overflow: -moz-scrollbars-none !important;
+        border-width: 0px !important;
+    }
 
-#liberator-completions {
-    -moz-user-focus: ignore;
-    overflow: -moz-scrollbars-none !important;
-    border-width: 0px !important;
-}
+    #liberator-bottombar:-moz-lwtheme {
+        text-shadow: none;
+    }
 
-#liberator-bottombar:-moz-lwtheme {
-    text-shadow: none;
-}
+    /* all elements in the statusline and commandline need some padding to look good */
+    #liberator-statusline {
+        font-family: monospace;
+        margin: 0px;
+        -moz-appearance: none;
+        border: none !important;
+        min-height: 16px !important;
+    }
+    #liberator-statusline:not([customizing="true"]) :-moz-any(toolbarbutton) {
+        border: none !important;
+        padding: 0 0px !important;
+        background: transparent !important;
+        height: 16px !important;
+    }
+    #liberator-statusline > *,
+    #liberator-commandline > * {
+        padding: 0px 2px;
+        font-family: monospace;
+    }
 
-/* all elements in the statusline and commandline need some padding to look good */
-#liberator-statusline {
-    font-family: monospace;
-    margin: 0px;
-    -moz-appearance: none;
-    border: none !important;
-    min-height: 16px !important;
-}
-#liberator-statusline:not([customizing="true"]) :-moz-any(toolbarbutton) {
-    border: none !important;
-    padding: 0 0px !important;
-    background: transparent !important;
-    height: 16px !important;
-}
-#liberator-statusline > *,
-#liberator-commandline > * {
-    padding: 0px 2px;
-    font-family: monospace;
-}
+    #liberator-commandline.hidden {
+        opacity: 0;
+        pointer-events: none; /* so we can click through when the command line is hidden */
+    }
 
-#liberator-commandline.hidden {
-    opacity: 0;
-    pointer-events: none; /* so we can click through when the command line is hidden */
-}
+    #liberator-statusline,
+    #liberator-message-toolbar {
+        -moz-box-align: center;
+    }
+    #liberator-statusline[customizing="true"] :-moz-any(#liberator-mode-toolbar, #liberator-message-toolbar, #liberator-status) {
+        border: 1px dotted gray;
+        min-width: 1em;
+        min-height: 1em;
+    }
+    #liberator-status > * {
+        padding: 0 1px;
+        margin: 0;
+    }
+    #liberator-status-ssl {
+        -moz-appearance: none;
+        width: 16px;
+        height: 16px;
+        visibility: collapse;
+        -moz-image-region: rect(0, 16px, 16px, 0);
+    }
+    #liberator-status-ssl.verifiedIdentity {
+        visibility: visible;
+        list-style-image: url(chrome://browser/skin/identity-icons-https-ev.png);
+    }
+    #liberator-status-ssl.verifiedDomain {
+        visibility: visible;
+        list-style-image: url(chrome://browser/skin/identity-icons-https.png);
+    }
+    #liberator-status-ssl.mixedActiveContent {
+        visibility: visible;
+        list-style-image: url(chrome://browser/skin/identity-icons-https-mixed-active.png);
+    }
 
-#liberator-statusline,
-#liberator-message-toolbar {
-    -moz-box-align: center;
-}
-#liberator-statusline[customizing="true"] :-moz-any(#liberator-mode-toolbar, #liberator-message-toolbar, #liberator-status) {
-    border: 1px dotted gray;
-    min-width: 1em;
-    min-height: 1em;
-}
-#liberator-status > * {
-    padding: 0 1px;
-    margin: 0;
-}
-#liberator-status-ssl {
-    -moz-appearance: none;
-    width: 16px;
-    height: 16px;
-    visibility: collapse;
-    -moz-image-region: rect(0, 16px, 16px, 0);
-}
-#liberator-status-ssl.verifiedIdentity {
-    visibility: visible;
-    list-style-image: url(chrome://browser/skin/identity-icons-https-ev.png);
-}
-#liberator-status-ssl.verifiedDomain {
-    visibility: visible;
-    list-style-image: url(chrome://browser/skin/identity-icons-https.png);
-}
-#liberator-status-ssl.mixedActiveContent {
-    visibility: visible;
-    list-style-image: url(chrome://browser/skin/identity-icons-https-mixed-active.png);
-}
+    #liberator-commandline-prompt {
+        background-color: inherit;
+        margin: 0px;
+    }
+    #liberator-commandline-command {
+        background-color: inherit;
+        color: inherit;
+        margin: 0px;
+    }
+    #liberator-message {
+        margin: 0px;
+    }
+    .liberator-hiding {
+        opacity: 0;
+        visibility: collapse;
+        transition-property: opacity, visibility;
+        transition-duration: .5s;
+    }
 
-#liberator-commandline-prompt {
-    background-color: inherit;
-    margin: 0px;
-}
-#liberator-commandline-command {
-    background-color: inherit;
-    color: inherit;
-    margin: 0px;
-}
-#liberator-message {
-    margin: 0px;
-}
-.liberator-hiding {
-    opacity: 0;
-    visibility: collapse;
-    transition-property: opacity, visibility;
-    transition-duration: .5s;
-}
+    #sidebar {
+        max-width: 90% !important;
+        min-width: 10% !important;
+    }
 
-#sidebar {
-    max-width: 90% !important;
-    min-width: 10% !important;
-}
+    /* MOW */
 
-/* MOW */
+    #liberator-completions,
+    #liberator-multiline-output,
+    #liberator-multiline-input {
+        overflow: hidden;
+        background-color: white;
+        color: black;
+    }
 
-#liberator-completions,
-#liberator-multiline-output,
-#liberator-multiline-input {
-    overflow: hidden;
-    background-color: white;
-    color: black;
-}
+    #liberator-completions-content,
+    #liberator-multiline-output-content,
+    #liberator-multiline-input {
+        white-space: pre;
+        font-family: -moz-fixed;
+        margin: 0px;
+    }
 
-#liberator-completions-content,
-#liberator-multiline-output-content,
-#liberator-multiline-input {
-    white-space: pre;
-    font-family: -moz-fixed;
-    margin: 0px;
-}
+    #liberator-completions-content *,
+    #liberator-multiline-output-content *,
+    #liberator-commandline-prompt *,
+    #liberator-commandline-command {
+        font: inherit;
+    }
 
-#liberator-completions-content *,
-#liberator-multiline-output-content *,
-#liberator-commandline-prompt *,
-#liberator-commandline-command {
-    font: inherit;
-}
+    #liberator-commandline-prompt-text {
+        max-width: 0;
+        transition-property: max-width;
+        transition-duration: 0.25s;
+    }
 
-#liberator-commandline-prompt-text {
-    max-width: 0;
-    transition-property: max-width;
-    transition-duration: 0.25s;
-}
+    #liberator-completions-content table,
+    #liberator-multiline-output-content table {
+        white-space: inherit;
+        border-spacing: 0px;
+    }
 
-#liberator-completions-content table,
-#liberator-multiline-output-content table {
-    white-space: inherit;
-    border-spacing: 0px;
-}
+    #liberator-completions-content td,
+    #liberator-multiline-output-content td,
+    #liberator-completions-content th,
+    #liberator-multiline-output-content th {
+        padding: 0px 2px;
+    }
 
-#liberator-completions-content td,
-#liberator-multiline-output-content td,
-#liberator-completions-content th,
-#liberator-multiline-output-content th {
-    padding: 0px 2px;
-}
-
-/* for muttator's composer */
-#content-frame, #appcontent {
-    border: 0px;
-}
-
+    /* for muttator's composer */
+    #content-frame, #appcontent {
+        border: 0px;
+    }
 }
 
 /* vim: set fdm=marker sw=4 ts=4 et: */

--- a/vimperator/install.rdf
+++ b/vimperator/install.rdf
@@ -15,8 +15,8 @@
         <em:targetApplication>
             <Description>
                 <em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
-                <em:minVersion>27.0</em:minVersion>
-                <em:maxVersion>35.0</em:maxVersion>
+                <em:minVersion>38.0</em:minVersion>
+                <em:maxVersion>50.0</em:maxVersion>
             </Description>
         </em:targetApplication>
     </Description>

